### PR TITLE
Use the editor font size for the breakpoint widget

### LINKED ISSUE #78790: Breakpoint widget should respect editor font size
#73685 isn't fixed

- VSCode Version: 1.37.0
- OS Version: Windows 10

> Steps to Reproduce:
>
> 1. Create a breakpoint
> 2. Right-click it and select "edit breakpoint"
> 3. See the text not vertically centred

Capture

### PR DESCRIPTION
**Pull request by anirudhrb**
Today, the breakpoint widget doesn't respect the `editor.fontSize`
setting. It always has the default font size even the user has increased the
font size for the editor. Modified `breakpointWidget.ts` to set the
`fontSize` option as per the setting.

Also fixed the centering of the input by calculating the top margin of the
input container dynamically:
> `topMargin = (heightOfContainer - lineHeight * numLines) / 2`

Tested by increasing the editor font size in settings and verified that the
breakpoint widget respects it. Also tested that a change in the setting 
reflects dynamically without having to re-open the widget.

Fixes #78790

### CODE REVIEW

File: src/vs/workbench/contrib/debug/browser/breakpointWidget.ts [hunk 1]
```diff
@@ -141,6 +146,7 @@ export class BreakpointWidget extends ZoneWidget implements IPrivateBreakpointWi
 	fitHeightToContent(): void {
 		const lineNum = this.input.getModel().getLineCount();
 		this._relayout(lineNum + 1);
+		this.centerInputVertically();
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/breakpointWidget.ts]: Does this end up calling `_doLayout` and does this end up calling then centerINputVertically twice?

File: src/vs/workbench/contrib/debug/browser/breakpointWidget.ts [hunk 2]
```diff
@@ -227,6 +236,29 @@ export class BreakpointWidget extends ZoneWidget implements IPrivateBreakpointWi
 				return suggestionsPromise;
 			}
 		}));
+
+		this.toDispose.push(this._configurationService.onDidChangeConfiguration((e) => {
+			if (e.affectsConfiguration('editor.fontSize')) {
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/breakpointWidget.ts]: If `editor.lineHeight` changes we can also update

File: src/vs/workbench/contrib/debug/browser/breakpointWidget.ts [hunk 3]
```diff
@@ -227,6 +236,29 @@ export class BreakpointWidget extends ZoneWidget implements IPrivateBreakpointWi
...
+		options.fontSize = editorConfig.fontSize;
+		return options;
+	}
+
+	private centerInputVertically() {
+		if (this.container) {
+			const lineHeight = this.input.getOption(EditorOption.lineHeight);
+			const lineNum = this.input.getModel().getLineCount();
+			const newTopMargin = (this.heightInPx - lineNum * lineHeight) / 2;
+			this.inputContainer.style.marginTop = newTopMargin + 'px';
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/breakpointWidget.ts]: I believe we can now remove the `margin-top` from the `breakpointWidget.css` is that right?

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
@anirudhrb thank you very much for your PR.
I have commented in the code directly. Can you please tackle my comments and once we polish this up we can merge it in.

File: src/vs/workbench/contrib/debug/browser/breakpointWidget.ts (hunk 3 above)
Comment (anirudhrb) [reply src/vs/workbench/contrib/debug/browser/breakpointWidget.ts]: Yeah, removed it now.

File: src/vs/workbench/contrib/debug/browser/breakpointWidget.ts (hunk 1 above)
Comment (anirudhrb) [reply src/vs/workbench/contrib/debug/browser/breakpointWidget.ts]: You are right. Removed this call to `centerInputVertically()`.

File: src/vs/workbench/contrib/debug/browser/breakpointWidget.ts (hunk 2 above)
Comment (anirudhrb) [reply src/vs/workbench/contrib/debug/browser/breakpointWidget.ts]: Done.

**Pull request merged**